### PR TITLE
Use Ninja instead of MSBuild on Windows

### PR DESCRIPTION
**Pull request by steven-johnson**
Note: this requires that we run the WinBot from cmd.exe (or maybe PowerShell) rather than MSYS2, as `vcvarsall` doesn't export the vars we need.

attn @alexreinking

### CODE REVIEW

**Comment by alexreinking** [comment]
PowerShell won't work straightaway. You would need to set the vars in CMD and then launch PowerShell. It sucks.

**Comment by alexreinking** [comment]
One more note: we will need to use the Visual Studio 16 2019 generator to produce release packages at least until CMake 3.17.4 when a bug I found in Ninja Multi-Config is fixed.

**Comment by abadams** [comment]
Given that it's changing, it would be nice to capture in a comment the procedure for starting the build worker from a fresh boot. I always have to guess at it or rely on shell history when I chromote into the workers.

**Comment by steven-johnson** [comment]
Good point. Will absolutely do so before this lands. (I plan on doing some testing before landing.)

**Comment by steven-johnson** [comment]
This still doesn't work, unfortunately: each ShellCommand we spawn is self-contained, so vcvarsall doesn't apply to subsequent steps. Probably need a step that will run vcvarsall, extract and return the env vars that are set (PATH, maybe others), and set them up in the env of subsequent steps.

**Comment by abadams** [comment]
I have just been using the msvc developer shell for windows work lately. It sucks as a shell but would be fine for running the buildbots under.

**Comment by abadams** [comment]
(my assumption is that subshells would inherit the environment)

**Comment by steven-johnson** [comment]
I think I have a fix that almost works now (and doesn't require cmd.exe) but ran out of time to finish debugging last night, and don't want to take over the buildbots to finish testing right now. Will resume once they catch up.

**Comment by steven-johnson** [comment]
Reworked and pretty sure this works now. PTAL

atten @alexreinking

File: master/master.cfg [hunk 1]
```diff
@@ -209,6 +211,90 @@ def add_get_source_steps(factory, llvm_branch):
...
+    "__DOTNET_PREFERRED_BITNESS",
+    "__VSCMD_PREINIT_PATH",
+    "__VSCMD_SCRIPT_ERR_COUNT",
+]
+
+
+def get_msvc_config_steps(factory, os):
+    if os.startswith('win'):
+        if '-32' in os:
+            arch = ' x86'
```
Comment (alexreinking) [inline master/master.cfg:268]: We might want this to be `x86_amd64` so we don't run out of memory in the linking step. I've found this is necessary for LLVM, but not (yet) for Halide.

Comment (alexreinking) [reply master/master.cfg]: Should be `amd64_x86`, oops.

Comment (abadams) [reply master/master.cfg]: To make sure I understand: This means it uses the 64-bit toolchain to produce 32-bit code (so using >2GB of ram is no big deal)?

Comment (steven-johnson) [reply master/master.cfg]: Yeah, nomenclature is confusing: https://docs.microsoft.com/en-us/cpp/build/building-on-the-command-line?view=vs-2019

**Comment by alexreinking** [comment]
Looks good to me

**Pull request merged**